### PR TITLE
ESQL: Adds a test and fixes running the CsvTestsDataLoader.main

### PR DESCRIPTION
**Pull request by astefan**
Following the changes in https://github.com/elastic/elasticsearch/pull/109552 running the `main` method of the csv loader (`gradlew :x-pack:plugin:esql:qa:testFixtures:loadCsvSpecData`) fails with `java.lang.NoClassDefFoundError: org/elasticsearch/xpack/esql/plan/logical/local/LocalRelation`.

This PR changes the gradle file and adds a test.

### CODE REVIEW

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/part-3

**Review by mark-vieira — COMMENTED** [review]
@breskeby can you sanity check this?

**Review by breskeby — APPROVED** [review]

**Comment by breskeby** [comment]
looking at that jarHell issue

File: x-pack/plugin/esql/qa/testFixtures/build.gradle [hunk 1]
```diff
@@ -1,9 +1,9 @@
 apply plugin: 'elasticsearch.java'
-
+apply plugin: org.elasticsearch.gradle.dependencies.CompileOnlyResolvePlugin
 
 dependencies {
     implementation project(':x-pack:plugin:esql:compute')
-    compileOnly project(':x-pack:plugin:esql')
+    api(project(':x-pack:plugin:esql'))
```
Comment (breskeby) [inline x-pack/plugin/esql/qa/testFixtures/build.gradle:6]: can we make that 'implementation'

File: x-pack/plugin/esql/qa/testFixtures/build.gradle [hunk 2]
```diff
@@ -14,6 +14,13 @@ dependencies {
     api "net.sf.supercsv:super-csv:${versions.supercsv}"
```
Comment (breskeby) [inline x-pack/plugin/esql/qa/testFixtures/build.gradle:14]: think this should be 'implementation' too

**Review by breskeby — CHANGES_REQUESTED** [review]
two minor remarks added. 

Also to fix the jarHell issue in `:xpack:plugin:esql` you need to tweak the testFixtures dependency declaration too exclude itself:

```
  testImplementation(project('qa:testFixtures')) {
    exclude(group:"org.elasticsearch.plugin", module: "esql")
  }
```

**Comment by astefan** [comment]
Thank you @breskeby. I've made the changes. Things (as in the `loadCsvSpecData` gradle task and the test I just added) work locally. Let's see if CI is happy as well.

**Review by breskeby — APPROVED** [review]

**Pull request merged**